pkg/schema/schema.go

```diff
@@ -167,6 +181,30 @@ func newCollectionItemValueType(collectionItemValue interface{}, position *filep
 	return nil, fmt.Errorf("Collection item type did not match any known types")
 }
 
+func processTypeAnnotation(templateAnnotations template.NodeAnnotations) (bool, error) {
```

Comment:
Since this function returns a bool, maybe rename it to indicate that. `hasAnyTypeAnnotation()`?